.github/workflows/snap.yml

```diff
@@ -0,0 +1,43 @@
+name: Publish Snap
+
+on:
+  push:
+    tags:
+      - 'v*.*.*'
+  workflow_dispatch: {}
+
+permissions:
+  contents: read
+
+jobs:
+  build-and-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      # speed-up LXD initialisation which snapcraft uses under the hood
+      - name: Setup LXD
+        run: sudo snap install lxd && sudo lxd waitready && sudo lxd init --auto
+
+      # Build the snap in the default channel, outputs the path of the .snap file
+      - name: Build snap
+        id: snapcraft
+        uses: snapcore/action-build@v1
+        with:
+          path: ./snap
```

Comment:
How does it know how to build the project?

Reply:
> It will use instructions in the ./snap/snapcraft.yaml to build the snap

I guess my question is how does it know how to build the project on the first place, as the YAML doesn't have specific CMake commands, etc